src/test/java/com/epam/dataservice/AccidentBatchLoaderTest.java

```diff
@@ -0,0 +1,197 @@
+    private static String filePath2 = "src/main/resources/DfTRoadSafety_Accidents_2010.csv";
+    private static String filePath3 = "src/main/resources/DfTRoadSafety_Accidents_2011.csv";
+    private static String filePath4 = "src/main/resources/DfTRoadSafety_Accidents_2012.csv";
+
+    public static void handle2FilesWithRunnable() {
+        ExecutorService executor = Executors.newFixedThreadPool(4);
+
+        BlockingQueue<List<RoadAccident>> dataQueue1 = new ArrayBlockingQueue<List<RoadAccident>>(2);
+        BlockingQueue<List<RoadAccident>> dataQueue2 = new ArrayBlockingQueue<List<RoadAccident>>(2);
+
+        AccidentBatchLoaderRunnable readerTask1 = new AccidentBatchLoaderRunnable(10000,dataQueue1,filePath1);
+        AccidentBatchLoaderRunnable readerTask2 = new AccidentBatchLoaderRunnable(10000,dataQueue2,filePath2);
+
+        executor.execute(readerTask1);
+        executor.execute(readerTask2);
+        System.out.println("Started readers");
+
+        executor.execute(new DataConsumer(dataQueue1, filePath1));
+        executor.execute(new DataConsumer(dataQueue2, filePath2));
+
+        System.out.println("started consumers");
+
+        endDatLoading(executor);
+
+    }
+
+    public static void handle2FilesWithCallable() throws ExecutionException, InterruptedException {
+        ExecutorService executor = Executors.newFixedThreadPool(4);
+
+        /**
+         * We want to process 2 files at a time so using 2 blocking queue
+         * to hold the result
+         */
+        BlockingQueue<List<RoadAccident>> dataQueue1 = new ArrayBlockingQueue<List<RoadAccident>>(2);
+        BlockingQueue<List<RoadAccident>> dataQueue2 = new ArrayBlockingQueue<List<RoadAccident>>(2);
+
+
+        //start consumer threads
+        executor.execute(new DataConsumer(dataQueue1, "dataQueue1"));
+        executor.execute(new DataConsumer(dataQueue2, "dataQueue2"));
+        System.out.println("Started Consumers");
+
+        //Create 2 reader task and start
+        FutureTask<Integer> readerTask1 = new FutureTask<Integer>(new AccidentBatchLoader(40000,dataQueue1,filePath1));
+        FutureTask<Integer> readerTask2 = new FutureTask<Integer>(new AccidentBatchLoader(40000,dataQueue2,filePath2));
+
+        System.out.println("Starting 2 readers");
+        executor.execute(readerTask1);
+        executor.execute(readerTask2);
+
+        readerTask1.get();
+        readerTask2.get();
+
+        endDatLoading(executor);
+    }
+
+    private static void endDatLoading(ExecutorService executor){
+        executor.shutdown();
+        try {
+            executor.awaitTermination(5, TimeUnit.MINUTES);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        System.out.println("Data loading finished");
+    }
+
+    public static void handleMultipleFilesWithCallable() throws ExecutionException, InterruptedException {
+
+        LinkedList<String> fileQueue = new LinkedList<>();
```

Comment:
Let's discuss this test tomorrow - it's too long and you are usign mutithreading here, when it's not required